WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.kt

```diff
@@ -300,17 +316,83 @@ class PostsListActivity : AppCompatActivity(),
         super.onCreateOptionsMenu(menu)
         menu?.let {
             menuInflater.inflate(R.menu.posts_list_toggle_view_layout, it)
-            val toggleViewLayoutMenuItem = it.findItem(R.id.toggle_post_list_item_layout)
+            toggleViewLayoutMenuItem = it.findItem(R.id.toggle_post_list_item_layout)
             viewModel.viewLayoutTypeMenuUiState.observe(this, Observer { menuUiState ->
                 menuUiState?.let {
                     updateMenuIcon(menuUiState.iconRes, toggleViewLayoutMenuItem)
                     updateMenuTitle(menuUiState.title, toggleViewLayoutMenuItem)
                 }
             })
+
+            searchActionButton = it.findItem(R.id.toggle_post_search)
+            initializeSearchView()
         }
         return true
     }
 
+    private fun initializeSearchView() {
+        if (!SiteUtils.isAccessedViaWPComRest(site)) {
+            searchActionButton.isVisible = false
+            return
+        }
+
+        searchActionButton.setOnActionExpandListener(object : OnActionExpandListener {
+            override fun onMenuItemActionExpand(item: MenuItem?): Boolean {
+                viewModel.onSearchExpanded(restorePreviousSearch)
+                return true
+            }
+
+            override fun onMenuItemActionCollapse(item: MenuItem?): Boolean {
+                viewModel.onSearchCollapsed()
+                return true
+            }
+        })
+
+        val searchView = searchActionButton.actionView as SearchView
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
```

Comment:
`onQueryTextChange` will not be invoked when you submit it (since there is no change to it). We need to handle `onQueryTextSubmit` for a couple of cases, eg., if you search for something and your internet connection turns off, you can submit a query with keyboard after your connection is back up without typing :)